lib/sentry/envelope.ex

```diff
@@ -34,6 +34,42 @@ defmodule Sentry.Envelope do
     }
   end
 
+  @doc """
+  Creates a new envelope containing the client report.
+  """
+  @doc since: "10.8.0"
+  @spec from_client_report(ClientReport.t()) :: t()
+  def from_client_report(%ClientReport{} = client_report) do
+    %__MODULE__{
+      event_id: UUID.uuid4_hex(),
+      items: [client_report]
+    }
+  end
+
+  @spec get_data_category(Attachment.t() | CheckIn.t() | ClientReport.t() | Event.t()) ::
+          String.t()
+  def get_data_category(type) do
```

Comment:
If you wanna raise, do it like this so that invalid input doesn't even get in the function:

```suggestion
  def get_data_category(%mod{} = type) when mod in [Attachment, CheckIn, ClientReport, Event] do
```